improver/fire_weather/fine_fuel_moisture_content.py

```diff
@@ -0,0 +1,367 @@
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+from typing import cast
+
+import numpy as np
+from iris.cube import Cube, CubeList
+
+from improver import BasePlugin
+
+
+class FineFuelMoistureContent(BasePlugin):
+    """
+    Plugin to calculate the Fine Fuel Moisture Code (FFMC) following
+    the Canadian Forest Fire Weather Index System.
+
+    The FFMC is a numerical rating of the moisture content of litter and other
+    fine fuels, representing the relative ease of ignition and flammability of fine fuel.
+    Values range from 0-101, with higher values indicating drier conditions.
+
+    This process is adapted directly from:
+        Equations and FORTRAN Program for the
+        Canadian Forest Fire Weather Index System
+        (C.E. Van Wagner and T.L. Pickett, 1985).
+        Page 5, Equations 1-10.
+
+    Expected input units:
+        - Temperature: degrees Celsius
+        - Precipitation: mm (24-hour accumulation)
+        - Relative humidity: percentage (0-100)
+        - Wind speed: km/h
+        - Previous FFMC: dimensionless (0-101)
+    """
+
+    temperature: Cube
+    precipitation: Cube
+    relative_humidity: Cube
+    wind_speed: Cube
+    input_ffmc: Cube
+    initial_moisture_content: np.ndarray
+    moisture_content: np.ndarray
+
+    def load_input_cubes(self, cubes: tuple[Cube] | CubeList):
+        """Loads the required input cubes for the FFMC calculation. These
+        are stored internally as Cube objects.
+
+        Args:
+            cubes (tuple[Cube] | CubeList): Input cubes containing the necessary data.
+
+        Raises:
+            ValueError: If the number of cubes does not match the expected
+                number (5).
+        """
+        names_to_extract = [
+            "air_temperature",
+            "lwe_thickness_of_precipitation_amount",
+            "relative_humidity",
+            "wind_speed",
+            "fine_fuel_moisture_content",
+        ]
+        if len(cubes) != len(names_to_extract):
+            raise ValueError(
+                f"Expected {len(names_to_extract)} cubes, found {len(cubes)}"
+            )
+
+        # Load the cubes into class attributes
+        (
+            self.temperature,
+            self.precipitation,
+            self.relative_humidity,
+            self.wind_speed,
+            self.input_ffmc,
+        ) = tuple(cast(Cube, CubeList(cubes).extract_cube(n)) for n in names_to_extract)
+
+        # Ensure the cubes are set to the correct units
+        self.temperature.convert_units("degC")
```

Comment:
I think 'celsius' is normally used here (rather than degC). It may work in the same way though?